### PR TITLE
Fix libc path for macOS El Capitan

### PR DESCRIPTION
**Pull request by jasonwilliams200OK**
Fix dotnet/cli#3875

/cc @jkotas

### CODE REVIEW

**Comment by jasonwilliams200OK** [comment]
I found similar fix was suggested here for El Capitan: ffi/ffi#461. I don't know why but I can't reproduce it locally with ditto version `System Version: OS X 10.11.5 (15F34)`. Good to have a fallback.

Even `DYLD_FALLBACK_LIBRARY_PATH=/usr/lib:$DYLD_FALLBACK_LIBRARY_PATH dotnet new` wouldn't work as it seems to explain what has changed in El Capitan: http://apple.stackexchange.com/a/212954. But interestingly, I also have:

```
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
```

like @PatNeedham: https://github.com/dotnet/cli/issues/3875#issuecomment-233135679, but it works for me.. Additionally, I have Xcode etc. installed so might be some other system configuration has it covered.

File: src/pal/src/loader/module.cpp [hunk 1]
```diff
@@ -1628,6 +1628,12 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     LockModuleList();
 
     dl_handle = LOADLoadLibraryDirect(shortAsciiName);
+#ifdef __APPLE__
+    if (!dl_handle)
```
Comment (jkotas) [inline src/pal/src/loader/module.cpp]: Shouldn't this fallback be limited to just failure to load libc?

Comment (jkotas) [inline src/pal/src/loader/module.cpp]: It is likely the reason why LoadLibrary tests are failing on OSX

**Comment by jkotas** [comment]
cc @janvorli

File: src/pal/src/loader/module.cpp (hunk 1 above)
Comment (jasonwilliams200OK) [inline src/pal/src/loader/module.cpp]: Ah you are right, I forgot that this method is *not* just loading libc. Will fix. :)

Comment (jasonwilliams200OK) [inline src/pal/src/loader/module.cpp]: Fixed in 9c716cf (by having a `bool isLibcRequested`).

File: src/pal/src/loader/module.cpp [hunk 2]
```diff
@@ -1628,6 +1636,12 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     LockModuleList();
 
     dl_handle = LOADLoadLibraryDirect(shortAsciiName);
+#ifdef __APPLE__
+    if (!dl_handle && isLibcRequested)
+    {
+        dl_handle = LOADLoadLibraryDirect("/usr/lib/libc.dylib");
```
Comment (jkotas) [inline src/pal/src/loader/module.cpp]: Is this really needed as fallback path? Would it be better to change the libc path above to `/usr/lib/libc.dylib` for Apple?

Comment (josteink) [inline src/pal/src/loader/module.cpp]: Seconding this comment. If this is to handle varying paths for different versions of OSX there has to be a better way, or some `#include` which can help us resolve this better? Either way I think the semantics and intentions behind the code could be improved. A simple comment might even do?

Comment (jasonwilliams200OK) [inline src/pal/src/loader/module.cpp]: @jkotas, I only have El Capitan to test this patch hence used the harmless fallback approach. If this is a known fact that all supported macOS versions will have `/usr/lib/libc.dylibc` as valid path, then I agree we should use just one path. Once confirmed, I will make the changed.

@josteink, in my second commit (7db7794), I fixed the outdated comment about libc. I can put one here as well. :) On macsOS and BSDs and Alpine, there is no equivalent of `#include <gnu/lib-names.h>` unfortunately.

Comment (janvorli) [inline src/pal/src/loader/module.cpp]: I would also prefer using the full path only on OSX. I have found the full path mentioned in **Mac OS X Internals: A Systems Approach** as the path of the libc. This book was printed in 2006, so it seems pretty stable:
https://books.google.cz/books?id=K8vUkpOXhN4C&pg=PA74&lpg=PA74&dq=libc.dylib+location&source=bl&ots=OLnnXYSwXy&sig=D1CMLFhuHQdMcnEw9J7Kq6KoVA8&hl=cs&sa=X&ved=0ahUKEwjL5o2vwP3NAhWKCMAKHRS0BpcQ6AEIXzAI#v=onepage&q=libc.dylib%20location&f=false

Comment (jasonwilliams200OK) [inline src/pal/src/loader/module.cpp]: Thanks @janvorli! Changed, squashed and re-pushed.

**Comment by janvorli** [comment]
LGTM

**Comment by janvorli** [comment]
Thank you!

**Pull request merged**